oggm/core/flowline.py

```diff
@@ -851,9 +852,25 @@ def step(self, dt):
             thick_stag[1:-1] = (thick[0:-1] + thick[1:]) / 2.
             thick_stag[[0, -1]] = thick[[0, -1]]
 
+            sf_func = None
+            sf = np.ones(thick.shape)  # Default shape factor is 1
+            if cfg.PARAMS['use_shape_factor_for_fluxbasedmodel'] == \
+                    'Adhikari' or \
+                    cfg.PARAMS['use_shape_factor_for_fluxbasedmodel'] == 'Nye':
+                sf_func = shape_factor_adhikari
+            elif cfg.PARAMS['use_shape_factor_for_fluxbasedmodel'] == 'Huss':
+                sf_func = shape_factor_huss
+
+            sf = sf_func(fl.widths_m, thick, fl.is_rectangular)
+            # TODO: avoid  unnecessary new array initialization by adding
+            # sf_stag to  self._stags
+            sf_stag = np.zeros(thick_stag.shape)
```

Comment:
Staggered shape factors need to be computed to match with `thick_stag` and `u_stag`.
I decided to use shape factors from the unstaggered grid and combine upstream and downstream shape factors to obtain staggered values.
Directly computing shape factors on the staggered grid would make an additional `width_stag` necessary and the question arises what `is_rectangular_stag` would look like.